python/zarrs/pipeline.py

```diff
@@ -32,10 +32,41 @@
 )
 
 
-@dataclass(frozen=True)
+def get_codec_pipeline_impl(codec_metadata_json: str) -> CodecPipelineImpl:
+    return CodecPipelineImpl(
+        codec_metadata_json,
+        validate_checksums=config.get(
+            "codec_pipeline.validate_checksums", None
+        ),
+        # TODO: upstream zarr-python array.write_empty_chunks is not merged yet #2429
+        store_empty_chunks=config.get("array.write_empty_chunks", None),
+        chunk_concurrent_minimum=config.get(
+            "codec_pipeline.chunk_concurrent_minimum", None
+        ),
+        chunk_concurrent_maximum=config.get(
+            "codec_pipeline.chunk_concurrent_maximum", None
+        ),
+        num_threads=config.get("threading.max_workers", None),
```

Comment:
Not sure even what the promise is from `zarr` re: the stability of values fetched from the `config` so not sure if this is a potential source of bugs or not